Python/Product/TestAdapter/ProjectInfo.cs

```diff
@@ -56,7 +56,8 @@ internal class ProjectInfo : IDisposable {
         public bool IsWorkspace => _pythonWorkspace != null;
 
         public TestContainer[] GetAllContainers() {
-            return _containers.Select(x => x.Value).ToArray();
+            //ConcurrentDictionary.ToArray() locks before copying
+            return _containers.Values.ToArray();
```

Comment:
Now ToArray() will lock the dictionary before it copies

Reply:
https://referencesource.microsoft.com/#mscorlib/system/Collections/Concurrent/ConcurrentDictionary.cs,bc495056297a0851

Reply:
Values locks and copies values and ToArray copies again